### PR TITLE
go-containerregistry: init at 0.4.1

### PR DESCRIPTION
**Pull request by yurrriq**
###### Motivation for this change

Add [go-containerregistry](https://github.com/google/go-containerregistry), which provides the `crane` and `gcrane` binaries.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
   - [x] `crane`
   - [ ] `gcrane` (I don't use this or have easy access to a GCR registry)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/go-containerregistry/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    rev = "v${version}";
+    hash = "sha256-3mvGHAPKDUmrQkBKwlxnF6PG0ZpZDqlM9SMkCyC5ytE=";
+  };
+
+  vendorSha256 = null;
+
+  inherit subPackages;
+
+  buildFlagsArray = [
+    "-ldflags=-s -w -X github.com/google/go-containerregistry/cmd/crane/cmd.Version=${version} -X github.com/google/go-containerregistry/pkg/v1/remote/transport.Version=${version}"
```
Comment (yurrriq) [inline pkgs/development/tools/go-containerregistry/default.nix:18]: is there some clever way to get/use `goPackagePath` here?

File: pkgs/development/tools/go-containerregistry/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{ lib, buildGoModule, fetchFromGitHub, subPackages ? [ "cmd/crane" "cmd/gcrane" ] }:
```
[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkgs/development/tools/go-containerregistry/default.nix:1]: Don't parameterize it at all and just build all packages.

File: pkgs/development/tools/go-containerregistry/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+buildGoModule rec {
+  pname = "go-containerregistry";
+  version = "0.4.1";
+  src = fetchFromGitHub {
```
Comment (SuperSandro2000) [inline pkgs/development/tools/go-containerregistry/default.nix:6]:
```suggestion
  version = "0.4.1";

  src = fetchFromGitHub {
```

File: pkgs/development/tools/go-containerregistry/default.nix (hunk 2 above)
[the remark this answers is not in the document]
Comment (yurrriq) [reply pkgs/development/tools/go-containerregistry/default.nix:1]: How does one override a `buildGoModule` expression then? For example, I will likely always want `subPackages = [ "cmd/crane" ]`.

File: pkgs/development/tools/go-containerregistry/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/tools/go-containerregistry/default.nix:19]: A let in could be used.

File: pkgs/development/tools/go-containerregistry/default.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  buildFlagsArray = [
+    "-ldflags=-s -w -X github.com/google/go-containerregistry/cmd/crane/cmd.Version=${version} -X github.com/google/go-containerregistry/pkg/v1/remote/transport.Version=${version}"
+  ];
+
+  # NOTE: no tests
+  doCheck = false;
+
+  meta = with lib; {
+    description = "A tool for interacting with remote images and registries";
+    inherit (src.meta) homepage;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/go-containerregistry/default.nix:27]:
```suggestion
    homepage = "https://github.com/google/go-containerregistry";
```

**Pull request merged**